node/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::if_same_then_else)]
```

Comment:
where do we have this case? are we sure it is not a mistake?

Reply:
There a few cases where this gets triggered, bcs it is used to dispatch one of actions, example:
```rs
pub fn transition_frontier_sync_ledger_snarked_success_effects<S: redux::Service>(
    _: &ActionMeta,
    store: &mut Store<S>,
) {
    if store.dispatch(TransitionFrontierSyncLedgerAction::Success) {
    } else if store.dispatch(TransitionFrontierSyncLedgerStagedAction::ReconstructEmpty) {
    } else if store.dispatch(TransitionFrontierSyncLedgerStagedAction::PartsFetchPending) {
    }
}
```